ENIGMAsystem/SHELL/Bridges/General/Win32-Direct3D9.cpp

```diff
@@ -223,40 +227,22 @@ namespace enigma_user {
 
 void display_reset(int samples, bool vsync) {
   if (d3dmgr == NULL) { return; }
-  IDirect3DSwapChain9 *sc;
-  d3dmgr->device->GetSwapChain(0, &sc);
   D3DPRESENT_PARAMETERS d3dpp;
-  sc->GetPresentParameters(&d3dpp);
-  if (vsync) {
-    d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_DEFAULT; // Present the frame immediately
-  } else {
-    d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_IMMEDIATE; // Present the frame immediately
-  }
-  d3dpp.MultiSampleType = (D3DMULTISAMPLE_TYPE)((int)D3DMULTISAMPLE_NONE + samples); // Levels of multi-sampling
-  d3dpp.MultiSampleQuality = 0; // No multi-sampling
-  if (samples) {
-    d3dmgr->SetRenderState(D3DRS_MULTISAMPLEANTIALIAS, TRUE);
-  } else {
-    d3dmgr->SetRenderState(D3DRS_MULTISAMPLEANTIALIAS, FALSE);
-  }
-  sc->Release();
+  get_d3d_present_params(&d3dpp);
+  d3dpp.PresentationInterval = vsync?D3DPRESENT_INTERVAL_DEFAULT:D3DPRESENT_INTERVAL_IMMEDIATE;
+  d3dpp.MultiSampleType = (D3DMULTISAMPLE_TYPE)((int)D3DMULTISAMPLE_NONE + samples);
+  d3dpp.MultiSampleQuality = 0;
+  d3dmgr->SetRenderState(D3DRS_MULTISAMPLEANTIALIAS, samples > 0);
 
   enigma::Reset(&d3dpp);
 }
 
 void set_synchronization(bool enable)
 {
   if (d3dmgr == NULL) { return; }
-  IDirect3DSwapChain9 *sc;
-  d3dmgr->device->GetSwapChain(0, &sc);
   D3DPRESENT_PARAMETERS d3dpp;
-  sc->GetPresentParameters(&d3dpp);
-  if (enable) {
-    d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_DEFAULT; // Present the frame immediately
-  } else {
-    d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_IMMEDIATE; // Present the frame immediately
-  }
-  sc->Release();
+  get_d3d_present_params(&d3dpp);
+  d3dpp.PresentationInterval = enable?D3DPRESENT_INTERVAL_DEFAULT:D3DPRESENT_INTERVAL_IMMEDIATE;
```

Comment:
Are we too good for spaces around our operators, now? 😛

Reply:
Yep, sorry, I seem to have taken a liking to the style now. ☹️